src/main/scala/org/allenai/plugins/DeployPlugin.scala

```diff
@@ -420,51 +456,87 @@ object DeployPlugin extends AutoPlugin {
+    * that should be shared between them, when deploying a project.
+    * @param hostConfigs configuration objects specifying how the project should be deployed to
+    *                    many hosts
+    * @param projectVersion the version of the project being deployed
+    */
+  case class DeployConfig(hostConfigs: Seq[HostConfig], projectVersion: Option[String])
+
+  /** Transform the given [[com.typesafe.config.Config]] object into a corresponding
+    * [[org.allenai.plugins.DeployPlugin.DeployConfig]] object.
     * @param targetName the name of the target to print on error
-    * @param targetConfig the config to parse
+    * @param targetConfig env-level deployment config to convert
     */
-  def validateAsMap(targetName: String, targetConfig: Config): Map[String, String] = {
-    // Validate that the target has all the required environment variables.
-    val requiredEnv = Seq("AWS_PEM_FILE")
-    for (variable <- requiredEnv) {
-      val value = sys.env.get(variable).getOrElse {
-        throw new IllegalArgumentException(s"Environment variable ${variable} undefined.")
+  def parseConfig(targetName: String, targetConfig: Config): DeployConfig = {
+
+    DeployConfig(
+      hostConfigs = Seq(parseHostConfig(targetConfig.getConfig("deploy"))),
+      projectVersion = try {
+        Some(targetConfig.getString("project.version"))
+      } catch {
+        case _: ConfigException.Missing => None
+        case _: ConfigException.WrongType =>
+          throw new IllegalArgumentException(s"Error: 'project.version' must be a string")
       }
+    )
+  }
 
-      require(!value.isEmpty, "Environment variable ${value} is empty.")
-    }
+  /** Transform the given [[com.typesafe.config.Config]] object into a corresponding
+    * [[org.allenai.plugins.DeployPlugin.HostConfig]] object.
+    * @param hostConfig host-level config to convert
+    */
+  def parseHostConfig(hostConfig: Config): HostConfig = {
+    val renderedConfig = hostConfig.root().render()
 
-    // Validate that the target has all the required keys.
-    val requiredKeys = Seq("deploy.host", "deploy.directory",
-      "deploy.startup_script", "deploy.user.ssh_username")
-    val requiredKeyPairs: Seq[(String, String)] = for {
-      key <- requiredKeys
-    } yield key -> {
-      try {
-        targetConfig.getString(key)
+    val requiredKeys = Seq("host", "directory", "startup_script", "user.ssh_username")
+
+    val confMap = (requiredKeys map { key =>
+      val value = try {
+        hostConfig.getString(key)
       } catch {
-        case configError: ConfigException =>
-          configError match {
-            case _: ConfigException.Missing =>
-              throw new IllegalArgumentException(s"Error: ${targetName} missing key ${key}.")
-            case _: ConfigException.WrongType =>
-              throw new IllegalArgumentException(s"Error: ${targetName}.${key} must be a string.")
-          }
+        case _: ConfigException.Missing =>
```

Comment:
Should we just let these exceptions bubble up rather than catching them and converting them to `IllegalArgumentException`s?  This is about as informative of an exception type as we'll be able to get without reinventing the wheel here.